### PR TITLE
[CIRC-1063] Allow creation of requests by overriding patron blocks

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: ramls/overrideBlocks.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Blocks to override (e.g. during checkout or renewal)",
+  "properties": {
+    "itemNotLoanableBlock": {
+      "description": "\"Item not loanable\" block",
```
Comment (alexanderkurash) [inline ramls/overrideBlocks.json:7]: Let's change " to '

File: ramls/overrideBlocks.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      "type": "object",
+      "dueDate": {
```
Comment (alexanderkurash) [inline ramls/overrideBlocks.json:9]: Should be inside "properties": {

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 3]
```diff
@@ -1946,6 +1966,156 @@ public void shouldNotCreateRequestWhenItemRequesterAndPickupServicePointCannotBe
...
+    createManualPatronBlockForUser(userId);
+    Response response = attemptCreateRequestThroughOverride(userId,
+      HEADERS_WITH_ALL_OVERRIDE_PERMISSIONS, null);
+
+    assertThat(response, hasStatus(HTTP_UNPROCESSABLE_ENTITY));
+    assertThat(response.getJson(), hasErrors(1));
+
+    assertThat(getErrorsFromResponse(response),
+      hasItem(
+        isBlockRelatedError("Patron blocked from requesting",
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:2059]: Same line?

**Comment by OleksandrVidinieiev** [comment]
@marcjohnson-kint could you please take a look at changes in chemas in this PR and see if they are aligned with your [latest comment in CIRC-1063](https://issues.folio.org/browse/CIRC-1063?focusedCommentId=96627&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-96627)?

File: ramls/extended-error.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the block",
+          "enum": [
+            "patronBlock",
+            "itemLimitBlock",
+            "itemNotLoanableBlock"
+          ]
+        },
```
Comment (marcjohnson-kint) [inline ramls/extended-error.json:35]: I think it is useful to clients that the API makes the relationship between an error and the mechanism to override it explicit

File: ramls/request.json [hunk 5]
```diff
@@ -433,6 +433,17 @@
       "description": "Metadata about creation and changes to requests, provided by the server (client should not provide)",
       "type": "object",
       "$ref": "raml-util/schemas/metadata.schema"
+    },
+    "requestProcessingParameters": {
+      "type": "object",
+      "description": "Additional parameters used for request processing and discarded afterwards. Not part of request record.",
```
Comment (marcjohnson-kint) [inline ramls/request.json:439]: Thank you for making it explicit that these properties are used for directing the process and not part of the record representation.

**Review by alexanderkurash — APPROVED** [review]

**Comment by OleksandrVidinieiev** [comment]
@marcjohnson-kint thanks for review!

**Pull request merged**